runbot_pylint/models/runbot_build.py

```diff
@@ -0,0 +1,217 @@
+                        get_depends(module_depends_str, addons_paths,
+                                    depends=depends)
+    return depends
+
+
+class RunbotBuild(models.Model):
+
+    """
+    Added pylint_conf_path field,
+    used by default the configuration of repository.
+    """
+
+    _inherit = "runbot.build"
+
+    pylint_conf_path = fields.Char(
+        help='Relative path to pylint conf file')
+
+    @api.model
+    def create(self, values):
+        """
+        This method set configuration of pylint.
+        """
+        if values.get('branch_id', False) \
+           and 'pylint_conf_path' not in values:
+            branch = self.env['runbot.branch'].browse(
+                values['branch_id'])
+            values.update({
+                'pylint_conf_path': branch.repo_id.pylint_conf_path,
+            })
+        return super(RunbotBuild, self).create(values)
+
+    @api.multi
+    def get_repo_branch_name(self):
+        """
+        This method get all repo id and branch name from a build.
+        Include dependency repo.
+        return dict {repo.id = branch_name}
+        """
+        repo_branch_data = {}
+        for build in self:
+            hint_branches = set()
+            for extra_repo in build.repo_id.dependency_ids:
+                closest_name = build.get_closest_branch_name(
+                    extra_repo.id, hint_branches)
+                hint_branches.add(closest_name)
+                repo_branch_data[extra_repo.id] = closest_name
+            repo_branch_data[build.repo_id.id] = build.name
+        return repo_branch_data
+
+    @api.multi
+    def get_modules_to_check_pylint(self):
+        """
+        This method get all modules to check pylint test.
+        Using field runbot_repo.check_pylint check box and get modules list
+        from branch with ls-tree.
+        This method use build.modules too to get all depends from
+        selected repo.
+        """
+        repo_pool = self.env['runbot.repo']
+        modules_to_check_pylint = set()
+        for build in self:
+            # get ls-tree modules from repo.check_pylint==True
+            repo_branch_name_data = build.get_repo_branch_name()
+            for repo_id in repo_branch_name_data:
+                repo = repo_pool.browse(repo_id)
+                if repo.check_pylint:
+                    branch_name = repo_branch_name_data[repo_id]
+                    branch_ls = repo.get_module_list(branch_name)
+                    modules_to_check_pylint |= set(branch_ls)
+
+            # get all depends and sub-depends from modules
+            _, modules = build.cmd()
+            depends = set(get_depends(modules, build.server('addons')))
+
+            # get all modules to check pylint intersection with modules depends
+            modules_to_check_pylint = list(depends & modules_to_check_pylint)
+        return modules_to_check_pylint
+
+    def job_15_pylint(self, cr, uid, build, lock_path, log_path, args=None):
```

Comment:
@bwrsandman 
I saw that `job_00_init` don't exists in [8.0](https://github.com/odoo/odoo-extra/blob/8.0/runbot/runbot.py#L751)
I will to revert to 15